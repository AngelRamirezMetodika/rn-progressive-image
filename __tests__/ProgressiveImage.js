import React from "react";
import { render } from "@testing-library/react-native";
import ProgressiveImage from "..";

describe("AnimatedProgressiveImage", () => {
    jest.useFakeTimers();
    const tree = (
        <ProgressiveImage
            small_source={{ uri: "https://i.imgur.com/T8bP9tw.jpg" }}
            large_source={{ uri: "https://i.imgur.com/v74Ah44.jpg" }}
            animation_library="animated"
        />
    );

    it("Animated API should render correctly", () => {
        const page = render(tree);
        expect(page.toJSON()).toMatchSnapshot();
    });
});

describe("ReanimatedProgressiveImage", () => {
    jest.useFakeTimers();
    const tree = (
        <ProgressiveImage
            small_source={{ uri: "https://i.imgur.com/T8bP9tw.jpg" }}
            large_source={{ uri: "https://i.imgur.com/v74Ah44.jpg" }}
            animation_library="reanimated"
        />
    );

    it("Reaimated API should render correctly", () => {
        const page = render(tree);
        expect(page.toJSON()).toMatchSnapshot();
    });
});
