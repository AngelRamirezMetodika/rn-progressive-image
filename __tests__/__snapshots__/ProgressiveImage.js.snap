// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedProgressiveImage Animated API should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "height": 375,
        "width": 375,
      },
      undefined,
    ]
  }
>
  <Image
    blurRadius={3}
    collapsable={false}
    nativeID="animatedComponent"
    source={
      Object {
        "uri": "https://i.imgur.com/T8bP9tw.jpg",
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <Image
    collapsable={false}
    nativeID="animatedComponent"
    onLoad={[Function]}
    source={
      Object {
        "uri": "https://i.imgur.com/v74Ah44.jpg",
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;

exports[`ProgressiveImage should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "height": 375,
        "width": 375,
      },
      undefined,
    ]
  }
>
  <Image
    blurRadius={3}
    collapsable={false}
    nativeID="animatedComponent"
    source={
      Object {
        "uri": "https://i.imgur.com/T8bP9tw.jpg",
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <Image
    collapsable={false}
    nativeID="animatedComponent"
    onLoad={[Function]}
    source={
      Object {
        "uri": "https://i.imgur.com/v74Ah44.jpg",
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;
